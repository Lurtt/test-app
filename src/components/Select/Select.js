import React from 'react'

export default function Select({ options, value, ...rest }) {
  return (
    <select {...rest} value={value ? value : 'placeholder'}>
      <option value="placeholder" disabled>
        Select your option
      </option>
      {options.map(option => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </select>
  )
}
