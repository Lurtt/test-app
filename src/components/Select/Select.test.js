import React from 'react'
import renderer from 'react-test-renderer'

import Select from './Select'

describe('<Select />', () => {
  test('Should render only one option', () => {
    const result = renderer.create(<Select options={[]} />).toJSON()
    expect(result).toMatchSnapshot()
  })
})
