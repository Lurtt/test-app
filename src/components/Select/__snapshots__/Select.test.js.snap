// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Should render only one option 1`] = `
<select
  className="sc-bdVaJa YQfmf"
  value="placeholder"
>
  <option
    disabled={true}
    value="placeholder"
  >
    Select your option
  </option>
</select>
`;
